src/bin/cargo-miri.rs

```diff
@@ -246,7 +247,9 @@ fn xargo_version() -> Option<(u32, u32, u32)> {
     Some((major, minor, patch))
 }
 
-fn ask_to_run(mut cmd: Command, ask: bool, text: &str) {
```

Comment:
I don't think we should forward the `MiriCommand` all the way to this helper function...  but I do agree with the idea of checking the env var in here, that in fact seems like the much better place than where you originally had that check.

Reply:
Fixed.